sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -47,15 +67,75 @@ object GpuMetricNames {
   val DESCRIPTION_NUM_PARTITIONS = "number of partitions"
   val DESCRIPTION_TOTAL_TIME = "total time"
   val DESCRIPTION_PEAK_DEVICE_MEMORY = "peak device memory"
-
-  def buildGpuScanMetrics(sparkContext: SparkContext): Map[String, SQLMetric] = {
+  val DESCRIPTION_COLLECT_TIME = "collect batch time"
+  val DESCRIPTION_CONCAT_TIME = "concat batch time"
+  val DESCRIPTION_SORT_TIME = "sort time"
+  val DESCRIPTION_AGG_TIME = "aggregation time"
+  val DESCRIPTION_JOIN_TIME = "join time"
+  val DESCRIPTION_JOIN_OUTPUT_ROWS = "join output rows"
+  val DESCRIPTION_FILTER_TIME = "filter time"
+  val DESCRIPTION_BUILD_DATA_SIZE = "build side size"
+  val DESCRIPTION_BUILD_TIME = "build time"
+  val DESCRIPTION_STREAM_TIME = "stream time"
+
+  // TODO these end up being ESSENTIAL_LEVEL do we really want that?
```

Comment:
Part of me thinks no GPU metric should be essential, only the Spark built-ins are, but I can see the argument that some GPU-specific metrics on some execs could be considered essential.  Curious what others think.

Reply:
I traced through a number of places for the metrics to see if they matched the Spark versions, and these I didn't find the time to do yet.  I'll look at them and see if I need to refactor it.